src/main/java/com/thealgorithms/backtracking/AllCombinations.java

```diff
@@ -0,0 +1,28 @@
+import java.util.*;
+
+public class AllCombinations {
+    public static void main(String[] args) {
+        int n = 5;
+        int k = 3;
+        List<List<Integer>> result = generateCombinations(n, k);
+        System.out.println(result);
+    }
+
+    public static List<List<Integer>> generateCombinations(int n, int k) {
+        List<List<Integer>> result = new ArrayList<>();
+        backtrack(result, new ArrayList<>(), 1, n, k);
```

Comment:
It is not compiling for me. I think instead of "new ArrayList<>()" as parameter, it should have "new ArrayList<Integer>()".

Reply:
made the required changes.